File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    result = []
    lottos.sort()
    win_nums.sort()
    if(max(lottos) == 0):
        result_0 = [1,6]
        return result_0
    else:
        count = 0 # 일치하는 개수
        for i in range(6):
            for j in range(6):
                if(lottos[i] == win_nums[j]): # 일치하는 개수 세기
                    count += 1
        count_0 = lottos.count(0) # 0 개수 세기 
        max_num = count + count_0
        result.append(rank(max_num))
        min_num = count
        result.append(rank(min_num))
        return result

def rank(n): # 순위
    if n >= 6:
        return 1
    elif n == 5:
        return 2
    elif n == 4:
        return 3
    elif n == 3:
        return 4
    elif n == 2:
        return 5
    else:
        return 6
'''
일부 번호를 알아볼 수 없게 되었다. 개수는 정해지 않음
구매했던 로또로 당첨이 가능했던 최고 순위와 최저 순위를 알아보고 싶다.
당첨 번호가 주어지고 

일단 lottos랑 win_nums 중 일치하는 개수 세고

'''

File: 프로그래머스/lv1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.java
class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
		int max = 0;
        
		for(int i= 0; i < 6; i++) {
			if(lottos[i] > max) {
				max = lottos[i]; // 최대값 찾기
			}
		}
		if(max == 0) { // 모두 0일 때
			int[] ans_0 = {1,6};
			return ans_0;
		}else { // 아닐 때
			int count = 0; // 일치하는 개수
			int count_0 = 0; // 0의 개수
			for(int i = 0; i < 6; i++) {
				for(int j = 0; j < 6; j++) {
					if(lottos[i] == win_nums[j]) {
						count++;
					}
				}
				if(lottos[i] == 0) {
					count_0++;
				}
			}
			int max_num = count + count_0;
			answer[0] = rank(max_num);
			answer[1] = rank(count);
			return answer;
		}
    }
    public static int rank(int n) {
		if(n>=6) {
			n = 6;
		}
		switch (n) {
		case 6:
			return 1; 
		case 5:
			return 2; 
		case 4:
			return 3; 
		case 3:
			return 4; 
		case 2:
			return 5; 
		default:
			return 6;
		}
	}
}